digitalocean/vpcpeering/resource_vpcpeering.go

```diff
@@ -0,0 +1,204 @@
+		}
+
+		_, _, err := client.VPCs.UpdateVPCPeering(context.Background(), d.Id(), vpcPeeringUpdateRequest)
+		if err != nil {
+			return diag.Errorf("Error updating VPC Peering: %s", err)
+		}
+		log.Printf("[INFO] Updated VPC Peering")
+	}
+
+	return resourceDigitalOceanVPCPeeringRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanVPCPeeringDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	vpcPeeringID := d.Id()
+
+	err := retry.RetryContext(ctx, d.Timeout(schema.TimeoutDelete), func() *retry.RetryError {
+		resp, err := client.VPCs.DeleteVPCPeering(context.Background(), vpcPeeringID)
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusForbidden {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(fmt.Errorf("error deleting VPC Peering: %s", err))
+		}
+
+		if err := waitForVPCPeeringDeletion(ctx, client, vpcPeeringID, d.Timeout(schema.TimeoutDelete)); err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		d.SetId("")
+		log.Printf("[INFO] VPC Peering deleted, ID: %s", vpcPeeringID)
+
+		return nil
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceDigitalOceanVPCPeeringRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	vpcPeering, resp, err := client.VPCs.GetVPCPeering(context.Background(), d.Id())
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			log.Printf("[DEBUG] VPC Peering (%s) was not found - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading VPC Peering: %s", err)
+	}
+
+	d.SetId(vpcPeering.ID)
+	d.Set("name", vpcPeering.Name)
+	d.Set("status", vpcPeering.Status)
+	d.Set("vpc_ids", vpcPeering.VPCIDs)
+	d.Set("created_at", vpcPeering.CreatedAt.UTC().String())
+
+	return nil
+}
+
+func validateVPCIDs(val []string) error {
+	if len(val) != 2 {
+		return fmt.Errorf("'vpc_ids' must contain exactly 2 VPC IDs, got %d", len(val))
+	}
+
+	vpcIDMap := make(map[string]struct{})
+	for _, id := range val {
+		vpcID := id
+		if _, ok := vpcIDMap[vpcID]; ok {
+			return fmt.Errorf("'vpc_ids' must contain unique VPC IDs, duplicate found: %s", vpcID)
+		}
+		vpcIDMap[vpcID] = struct{}{}
+	}
+
+	return nil
+}
+
+func waitForVPCPeeringDeletion(ctx context.Context, client *godo.Client, vpcPeeringID string, timeout time.Duration) error {
+	deadline := time.Now().Add(timeout)
+	for {
+		_, resp, err := client.VPCs.GetVPCPeering(ctx, vpcPeeringID)
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusNotFound {
+				// VPC peering is fully deleted
+				return nil
+			}
+			return fmt.Errorf("error checking VPC Peering status: %s", err)
+		}
+
+		if time.Now().After(deadline) {
+			return fmt.Errorf("timed out waiting for VPC Peering to be deleted")
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+}
```

Comment:
probably better to do it this way https://github.com/digitalocean/terraform-provider-digitalocean/blob/main/digitalocean/loadbalancer/resource_loadbalancer.go#L698